plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-webhook-handler.php

```diff
@@ -145,13 +145,22 @@ private function capture_payment( $order, $links ) {
 			}
 		}
 
-		$payment_gateways = WC()->payment_gateways()->payment_gateways();
-		if ( ! isset( $payment_gateways['paypal'] ) ) {
-			WC_Gateway_Paypal::log( 'PayPal gateway is not available.' );
+		if ( ! $capture_url ) {
+			WC_Gateway_Paypal::log( 'No capture URL found in webhook links', 'error' );
 			return;
 		}
-		$gateway        = $payment_gateways['paypal'];
-		$paypal_request = new WC_Gateway_Paypal_Request( $gateway );
-		$paypal_request->capture_payment( $order, $capture_url );
+
+		// Schedule a background job to capture the payment.
+		WC()->queue()->schedule_single(
+			time() + 10, // Capture in 10 seconds.
```

Comment:
I'm not familiar with action scheduler. Does this mean it will run 10 seconds from when the webhook was received? Or does it need additional traffic to trigger the function we hooked into `init`?

Reply:
<img width="247" height="372" alt="Screenshot 2025-07-25 at 12 15 19 AM" src="https://github.com/user-attachments/assets/0f0fe336-ffb8-4167-bdd7-10d3f2791c5d" />

Looks like it's the latter 🤔 which makes me worried about low-traffic stores and if there's a capture time limit. Not sure what to do, though.